bowling/tests/Game.ut.cpp

```diff
@@ -0,0 +1,31 @@
+#include <gtest/gtest.h>
+#include "Game.hpp"
+
+constexpr char inputDirectoryPath[] = "../input/";
+constexpr char resultOutput[] = "../tests/resultTest.txt";
+
+TEST(GameTest, ResultShouldBeEqualToInitialFile) {
+    Game game{inputDirectoryPath, resultOutput};
+    game.saveDataToFile();
+    std::string expected = "### Lane 1: game in progress ###"
+                           "Name1 18"
+                           "Name2 17"
+                           " 24"
+                           "### Lane 2: no game ###"
+                           "### Lane 3: game finished ###"
+                           "Michael 167"
+                           "Radek 90";
+
+    std::ifstream testFile(resultOutput, std::ios_base::in);
+    std::stringstream os;
+    std::string line;
+    if(testFile.is_open()){
+        while(getline(testFile,line)){
+            os << line;
+        }
+        testFile.close();
+    } else {
+        std::cout << "Cannot open file\n";
+    }
+    ASSERT_EQ(os.str(), expected);
+}
```

Comment:
What about empty directories, dirs with one or many files, with empty files? They may be tested in DirectoryHandler but here some general test of some system should be provided. You can mock some functions to return proper values if you don't want to use prepared files (it is recommended way).

Reply:
Tests added by using prepared files and directories, unfortunately github don't allow to add empty directory - so tests which dealing with empty directory I commented out to pass github CI.